iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift

```diff
@@ -72,7 +72,15 @@ class ConversationMessagesCell: UICollectionViewCell {
     func set(sequence: MessageSequence) {
         self.conversation = sequence
 
-        self.dataSource.update(messageSequence: sequence)
+        self.dataSource.set(messageSequence: sequence)
+    }
+
+    func set(layoutForDropZone: Bool) {
+        self.collectionLayout.layoutForDropZone = layoutForDropZone
+    }
+
+    func set(isPreparedToSend: Bool) {
+        self.dataSource.shouldPrepareToSend = isPreparedToSend
```

Comment:
To improve cell reuse behavior, set a persistent variable on the datasource and layout objects for send preparation.

Now we won't lose the preparation UI when scrolling around.